python/test/unit/language/test_core.py

```diff
@@ -2966,6 +2966,26 @@ def histogram_kernel(x_ptr, z_ptr, M: tl.constexpr, N: tl.constexpr):
     assert (z_torch == z).all()
 
 
+@pytest.mark.interpreter
+def test_histogram_complex_layout(device):
+
+    @triton.jit
+    def test(data, counts, SIZE: tl.constexpr):
+        data = tl.load(data + tl.arange(0, SIZE * SIZE)).reshape(SIZE, SIZE)
+        result = tl.dot(data, data).argmin(axis=1)
```

Comment:
it would be better to write a gluon test so that you can pick the layout explicitly

Reply:
you cannot mix triton and Gluon but Gluon will exercise this code so writing a pure Gluon test is better

Reply:
the big advantage there is that you can pick the layout so you can test all the corner cases of data replicated within warps and across warps explicitly

Reply:
https://github.com/triton-lang/triton/pull/7989